docs/proposals/202105256-cluster-class-and-managed-topologies.md

```diff
@@ -799,6 +879,14 @@ intentionally use resources without patches and variables to focus on the simple
        ```
       Note: The topology label needs to be set on the individual Machine objects as well.
     1. Creates the Machine Deployment object in the API server.
+6. MachineHealthCheck object creation
+   1. If `ClusterClass.spec.machineHealthCheck` is set:
+      1. For each `Cluster.spec.topology.workers.machineDeployments` item in the list create a MachineHealthCheck with the label `topology.cluster.x-k8s.io/deployment-name: <machine-deployment-topology-name>` as the selector and set the `clusterName` field from the cluster. 
+      2. For the Cluster.spec.topology.controlPlane create a MachineHealthCheck with the label `cluster.x-k8s.io/control-plane` as the selector and set the `clusterName` field from the cluster.
+   2. For each `Cluster.spec.topology.workers.machineDeployments` with `ClusterClass.spec.workers.machineDeployments[i].machineDeploymentClass` defined create a MachineHealthCheck with the label `topology.cluster.x-k8s.io/deployment-name: <machine-deployment-topology-name>` as the selector and set the `clusterName` field from the cluster.
+   3. If `Cluster.spec.topology.controlPlane.machineHealthCheckClass` is set create a MachineHealthCheck with the label `cluster.x-k8s.io/control-plane` as the selector and set the `clusterName` field from the cluster.
+
+**Note** `ClusterClass.spec.workers.machineDeployments[i].machineDeploymentClass` and `Cluster.spec.topology.controlPlane.machineHealthCheckClass` can not be set if `Cluster.spec.topology.workers.machineDeployments` is set.
```

Comment:
Maybe we should follow the same structure as above with the InfraCluster/ControlPlane/MD creation or even embed it in that logic that it would only be an additional step during  InfraCluster/ControlPlane/MD reconciliation

Reply:
I've add the If NOT line as the simpler of the two options - let me know if this works.

Reply:
Fine for me!